diff --git a/xls/build_rules/xls_ir_rules.bzl b/xls/build_rules/xls_ir_rules.bzl
index b5cd2dce1..e48747e2b 100644
--- a/xls/build_rules/xls_ir_rules.bzl
+++ b/xls/build_rules/xls_ir_rules.bzl
@@ -1174,6 +1174,7 @@ def _xls_ir_cc_library_impl(ctx):
             unformatted = unformatted_header_file.path,
             formatted = header_file.path,
         ),
+        use_default_shell_env = True,
         toolchain = None,
     )
 
@@ -1187,6 +1188,7 @@ def _xls_ir_cc_library_impl(ctx):
             unformatted = unformatted_source_file.path,
             formatted = source_file.path,
         ),
+        use_default_shell_env = True,
         toolchain = None,
     )
 
